test/cmdlineTests.sh

```diff
@@ -38,7 +38,20 @@ source "${REPO_ROOT}/scripts/common.sh"
 source "${REPO_ROOT}/scripts/common_cmdline.sh"
 
 AUTOUPDATE=false
-[[ $1 == --update ]] && AUTOUPDATE=true && shift
+no_smt=false
+while [[ $# -gt 0 ]]
```

Comment:
If we do this, we can drop the part later in this file that reports an error when `--update` is mixed in with other arguments (line 313).

Reply:
done

Reply:
Ah, cmdlineTests get test names as argument. In the first version I didn't support that. So I add another variable, `tdirs`, to store list of received test names.
by the way I don't know how this piece of code: `${*:-*/}` works :)
please take another look.

Reply:
> by the way I don't know how this piece of code: `${*:-*/}` works :)

Magic :P See https://github.com/ethereum/solidity/pull/11712#discussion_r678289856.